### ISSUE TITLE
Size resetting after restart

### ISSUE THREAD

**Issue by Rapha149** [mauritzn/Web-Viewer-KDE-Widget#1]
The size of the widget resets for me after a logging in again.
It's because of these lines:
```qml
// Defaults should probably be changed, but the widget isn't really intended to run at these sizes,
// it's just here to make sure it doesn't start as a tiny square.
Layout.preferredWidth: 640
Layout.preferredHeight: 360
```

I would rather have a widget that is small and that I have to resize one time than having to resize the widget every time I login.
So option 1 would be to just delete these lines.

Option 2 is to add the size to the configuration as described here: https://develop.kde.org/docs/plasma/widget/examples/#configurable-panel-widget-widthheight

**Comment by mauritzn** [comment]
I never expected an issue submission for this repo, mainly since I didn't expect anyone to find it, but thank you for checking it out.

I don't remember encountering this issue back when I developed it, but I was using a rather old version of KDE at the time and also decommissioned the use of the widget fairly early due to not having much use for it. This was my first time ever messing around with QML and KDE widgets and the whole thing was made in the course of a day so I suspected it would have some issues.

I currently do not use KDE anymore since I moved over to using Cinnamon, as such I don't have a way of testing this widget in a native environment anymore. But I'll fire up a VM and look into option 2 and if that gets too problematic I'll implement option 1.

Thank you again for taking the time to find this issue and reporting it.

**Comment by Rapha149** [comment]
> I never expected an issue submission for this repo, mainly since I didn't expect anyone to find it, but thank you for checking it out.  

I know the feeling :upside_down_face: 

Thanks for looking into it!

**Comment by mauritzn** [comment]
I've pushed an update to the main branch, it should fix the issue

**Comment by Rapha149** [comment]
It doesn't work for me.
You forgot to add the necessary code in the `ConfigGeneral.qml`:
```qml
property alias cfg_width: width.value
property alias cfg_height: height.value
```
and
```qml
RowLayout {
    Kirigami.FormData.label: i18n("Widget size:")
    SpinBox {
        id: width
        from: 0
        to: 2147483647 // 2^31-1
    }
    Label {
        text: " x "
    }
    SpinBox {
        id: height
        from: 0
        to: 2147483647 // 2^31-1
    }
}
```

But even with that it doesn't work for me :shrug:

**Comment by mauritzn** [comment]
Odd the updated code worked fine for me with multiple differently sized widgets. All of them remained the correct size and position after both relogs and reboots _(Kubuntu 23.04)_. Are you sure your running the latest widget code? I think you need to remove all active versions of the widget and replace the widget in `/home/<USER>/.local/share/plasma/plasmoids` with the updated version, then do a reboot if you want to ensure correct reloading.

I didn't add the width and height properties in the settings panel since I thought it would be better to just let the user drag resize it, but guess I can add that as well.

**Comment by mauritzn** [comment]
After adding it to the settings panel as well it still works as expected, e.g. I can change the size of the widget either by drag-resizing or by using the settings panel and it remembers the size correctly. But the settings panel is quite inconsistent with updating it's values properly so I much prefer the drag-resizing when in widget edit mode.

What OS/KDE version are you using? Since it seems to be something on your end that is causing this issue.

**Comment by Rapha149** [comment]
Ah, I didn't know it was supposed to work that way. But it still doesn't work for me, even after multiple restarts and reinstallation attempts.

I use OpenSUSE Leap 15.4 and KDE Plasma 5.24.4

But I guess I will just make it work by using option 1 for me locally.

**Comment by mauritzn** [comment]
> But I guess I will just make it work by using option 1 for me locally.

Seems that way, considering how simple the code is it should be easy enough to modify it to suit your needs.

But either way I'll try making a VM using OpenSUSE Leap 15.4 tomorrow and see if I can reproduce this issue.

**Comment by mauritzn** [comment]
Had some time to get a OpenSUSE VM up and running today and after some testing I did have some issues in OpenSUSE. I have pushed a patch to the main branch that seems to resolve it. I still haven't included the width & height options in the settings panel since they are still a bit odd and inconsistent.

The main issue seems to be that OpenSUSE's KDE for some reason ignores saved sizes if they were smaller than the default value in the config file, as such I lowered the default size to 150x150 and the issue seems to have been resolved.

**Here's how I resize the widget _(in widget edit mode)_:**
![Screenshot_20230612_232016_cropped](https://github.com/mauritzn/Web-Viewer-KDE-Widget/assets/11979966/cf90b346-c14c-46f3-a184-c79ec490fa57)

Hope this fixes the issue for you, if you are still having issues then I'm not sure what I can do on my end, since my OpenSUSE install is no longer having the issue.
